crates/kornia-core-ops/src/ops.rs

```diff
@@ -1,78 +1,203 @@
 use kornia_core::{storage::TensorStorage, SafeTensorType, Tensor, TensorAllocator};
+use num_traits::Float;
 
 use crate::error::TensorOpsError;
 
-/// Compute the sum of the elements in the tensor along dimension `dim`
-///
-/// # Arguments
-///
-/// * `tensor` - The tensor to sum the elements of.
-/// * `dim` - The index of the dimension/axis to perform the sum operation over.
-///
-/// # Returns
-///
-/// A new `Tensor` containing the element sums.
-///
-/// # Errors
-///
-/// If the requested dimension is greater than the number of axes of the tensor, an error is returned.
-///
-/// # Example
-///
-/// ```
-/// use kornia_core::{Tensor, CpuAllocator};
-/// use kornia_core_ops::ops::sum_elements;
-///
-/// let data: [u8; 6] = [1, 1, 1, 1, 1, 1];
-/// let t = Tensor::<u8, 2>::from_shape_slice([2, 3], &data, CpuAllocator).unwrap();
-/// let agg = sum_elements(&t, 1).unwrap();
-/// assert_eq!(agg.shape, [2, 1]);
-/// assert_eq!(agg.as_slice(), [3, 3]);
-/// ```
-pub fn sum_elements<T, const N: usize, A>(
-    tensor: &Tensor<T, N, A>,
-    dim: usize,
-) -> Result<Tensor<T, N, A>, TensorOpsError>
+/// A trait to define operationt that can be performed on a tensor.
+pub trait TensorOps<T, const N: usize, A>
 where
-    T: SafeTensorType + std::ops::Add<Output = T>,
-    A: TensorAllocator + Clone + 'static,
+    T: SafeTensorType,
+    A: TensorAllocator,
 {
-    if dim >= N {
-        return Err(TensorOpsError::DimOutOfBounds(dim, N - 1));
-    }
+    /// Apply the power function to the tensor data.
+    ///
+    /// # Arguments
+    ///
+    /// * `n` - The power to raise the tensor data to.
+    ///
+    /// # Returns
+    ///
+    /// A new tensor with the data raised to the power.
+    fn powf(&self, n: T) -> Tensor<T, N, A>
+    where
+        T: Float;
+
+    /// Apply the square root function to the tensor elements.
+    ///
+    /// # Returns
+    ///
+    /// A new tensor with the square root values.
+    fn sqrt(&self) -> Tensor<T, N, A>
+    where
+        T: Float;
+
+    /// Compute the sum of the elements in the tensor along dimension `dim`
+    ///
+    /// # Arguments
+    ///
+    /// * `dim` - The index of the dimension/axis to perform the sum operation over.
+    ///
+    /// # Returns
+    ///
+    /// A new `Tensor` containing the element sums.
+    ///
+    /// # Errors
+    ///
+    /// If the requested dimension is greater than the number of axes of the tensor, an error is returned.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use kornia_core::{Tensor, CpuAllocator};
+    /// use kornia_core_ops::ops::TensorOps;
```

Comment:
```suggestion
    /// use kornia_core_ops::TensorOps;
```
Maybe we can expose to root ?